src/scrapers/beinleumi.js

```diff
@@ -0,0 +1,265 @@
+    description: getTransactionDescription(tds, transactionsColsTypes),
+    reference: getTransactionReference(tds, transactionsColsTypes),
+    debit: getTransactionDebit(tds, transactionsColsTypes),
+    credit: getTransactionCredit(tds, transactionsColsTypes),
+  };
+}
+
+async function getTransactionsColsTypeClasses(page) {
+  const typeClassesMap = [];
+  const typeClassesObjs = await pageEvalAll(page, `${TRANSACTIONS_TABLE} tbody tr:first-of-type td`, null, (tds) => {
+    return tds.map((td, index) => ({
+      colClass: td.getAttribute('class'),
+      index,
+    }));
+  });
+
+  for (const typeClassObj of typeClassesObjs) {
+    typeClassesMap[typeClassObj.colClass] = typeClassObj.index;
+  }
+  return typeClassesMap;
+}
+
+function extractTransaction(txns, txnRow, transactionsColsTypes) {
+  const txn = extractTransactionDetails(txnRow, transactionsColsTypes);
+  if (txn.date !== '') {
+    txns.push(txn);
+  }
+}
+
+async function extractTransactions(page) {
+  const txns = [];
+  const transactionsColsTypes = await getTransactionsColsTypeClasses(page);
+
+  const transactionsRows = await pageEvalAll(page, `${TRANSACTIONS_TABLE} tbody tr`, [], (trs) => {
+    return trs.map((tr) => ({
+      innerTds: Array.from(tr.getElementsByTagName('td')).map((td) => td.innerText),
+    }));
+  });
+
+  for (const txnRow of transactionsRows) {
+    extractTransaction(txns, txnRow, transactionsColsTypes);
+  }
+  return txns;
+}
+
+async function isNoTransactionInDateRangeError(page) {
+  const hasErrorInfoElement = await elementPresentOnPage(page, `.${ERROR_MESSAGE_CLASS}`);
+  if (hasErrorInfoElement) {
+    const errorText = await page.$eval(`.${ERROR_MESSAGE_CLASS}`, (errorElement) => {
+      return errorElement.innerText;
+    });
+    return errorText.trim() === NO_TRANSACTION_IN_DATE_RANGE_TEXT;
+  }
+  return false;
+}
+
+async function searchByDates(page, startDate) {
+  await clickButton(page, 'a#tabHeader4');
+  await waitUntilElementFound(page, 'div#fibi_dates');
+  await fillInput(
+    page,
+    'input#fromDate',
+    startDate.format(DATE_FORMAT),
+  );
+  await clickButton(page, `button[class*=${CLOSE_SEARCH_BY_DATES_BUTTON_CLASS}]`);
+  await clickButton(page, `input[value=${SHOW_SEARCH_BY_DATES_BUTTON_VALUE}]`);
+  await waitForNavigation(page);
+}
+
+async function getAccountNumber(page) {
+  const selectedSnifAccount = await page.$eval(ACCOUNTS_NUMBER, (option) => {
+    return option.innerText;
+  });
+
+  return selectedSnifAccount.replace('/', '_');
+}
+
+async function navigateToNextPage(page) {
+  await clickButton(page, NEXT_PAGE_LINK);
+  await waitForNavigation(page);
+}
+
+async function scrapeTransactions(page) {
+  const txns = [];
+  for (let i = 0; i < TRANSACTIONS_PAGINATION_LIMIT; i += 1) {
+    txns.push(...await extractTransactions(page));
+    try {
+      await navigateToNextPage(page);
+    } catch (error) {
+      if (error.message.includes(CLICK_ELEMENT_NOT_EXISTS_ERROR_MSG)) {
```

Comment:
@dudiventura could you explain why you're breaking based on this error?

Reply:
Hi @eshaham,
Yes, I've tried first to use the function **elementPresentOnPage** to check if the "next page" link element presents in the page.
But for some reason I kept getting a positive answer (true) from the function even when the element wasn't there.

And when the script tries to click on element using the function **clickButton** while the element doesn't exists I get the error "Cannot read property".

So when this error is thrown, I catch it, validate the this is the error and by that assume that the element doesn't exists in the page and there is no next page so I can break the loop there.

I know it's not the most elegant way, so if you have a suggestion to another way I'll be happy to learn something new.

Reply:
hmm... the method `elementPresentOnPage` should be reliable. just to make sure you have tried:
1. pausing the browser and making sure the selector is correct.
```
await page.evaluate(() => {
  debugger;
});
```
2. calling `elementPresentOnPage` with await.

Reply:
Thank you for your comment, I'm checking it again.

Reply:
Hi @esakal,
You are very right, I had a missing "await" in the code and therefore I didn't get the correct answer from the function.
I'm deploying a fix now.